lib/dialects/abstract/query-generator.js

```diff
@@ -1170,6 +1170,16 @@ class QueryGenerator {
           attributes.main.push(mainTable.model.rawAttributes[keyAtt].field ? [keyAtt, mainTable.model.rawAttributes[keyAtt].field] : keyAtt);
         }
       }
+
+      // appending fields which are required by 'BelongsTo' associations when main attributes filter is in use
+      if (options.include) {
+        for (const include of options.include) {
+          const { associationType, foreignKey } = include.association;
```

Comment:
I am not very familiar with this part of the code, how can we be 100% sure that `include.association` will always be defined here (and be an object)? Recall that `const { x, y } = undefined;` gives `TypeError: Cannot destructure property `x` of 'undefined' or 'null'`. I would be more confident if this was wrapped in an `if (include.association)`... What do you think?

Reply:
Good point. Updated it to _paranoid_ mode

Reply:
@wertlex Excellent!